build/transpile.js

```diff
@@ -702,6 +714,16 @@ class Transpiler {
             allVariables.push (catchClauseMatches[1])
         }
 
+        // match all variables instantiated as function parameters
```

Comment:
This is wrong, since it will make them global per entire file. Those variable should be local to a function in question. Otherwise, if you have two functions in a file those variables will be mixed and confused. We might have to rewrite the transpiler substantially in order to handle the function scope properly...

Reply:
In order to handle the function scope correctly, we may have to make a recursive transpile method. However, let's not do that yet. Just wanted to know your thoughts on this subject. Maybe we should start adopting the AST and rewrite the transpiler for it... The AST approach can be slower, though.